lib/config/applyrule.cpp

```diff
@@ -75,7 +75,9 @@ void ApplyRule::AddRule(const String& sourceType, const String& targetType, cons
 	const Expression::Ptr& expression, const Expression::Ptr& filter, const String& package, const String& fkvar,
 	const String& fvvar, const Expression::Ptr& fterm, bool ignoreOnError, const DebugInfo& di, const Dictionary::Ptr& scope)
 {
-	m_Rules[sourceType].push_back(ApplyRule(targetType, name, expression, filter, package, fkvar, fvvar, fterm, ignoreOnError, di, scope));
+	m_Rules[Type::GetByName(sourceType).get()][Type::GetByName(targetType).get()].emplace_back(ApplyRule(
+		targetType, name, expression, filter, package, fkvar, fvvar, fterm, ignoreOnError, di, scope
+	));
```

Comment:
What's the point of `emplace_back` if you call the constructor in the argument anyways?

Reply:
If I can use emplace_back, I use emplace_back.

Reply:
But wouldn't it make more sense to then use `emplace_back(x,yz)` instead of `emplace_back(T(x,y,z))`? Apart from that, my favorite reading advice in that regard: https://abseil.io/tips/112

Reply:
Sure, but the constructor is private.